src/cpp/src/cache_manager.hpp

```diff
@@ -156,14 +160,12 @@ class CacheManager {
                 update_request_tensor(decoder_layer_id);
             }
         } else {
-            auto remote_context = m_request.get_compiled_model().get_context();
-
             for (size_t decoder_layer_id = 0; decoder_layer_id < m_num_decoder_layers; ++decoder_layer_id) {
                 ov::Shape value_cache_shape = set_kv_blocks(m_value_shapes[decoder_layer_id], num_kv_blocks);
                 ov::Shape key_cache_shape = set_kv_blocks(m_key_shapes[decoder_layer_id], num_kv_blocks);
 
-                ov::Tensor key_cache = remote_context.create_tensor(get_key_cache_precision(decoder_layer_id), key_cache_shape);
-                ov::Tensor value_cache = remote_context.create_tensor(get_value_cache_precision(decoder_layer_id), value_cache_shape);
+                ov::Tensor key_cache = m_context.create_tensor(get_key_cache_precision(decoder_layer_id), key_cache_shape);
```

Comment:
should not the condition `if (m_device.find("GPU") == std::string::npos)` above be changed to `if (m_context)` ?

Reply:
Yes, `if (m_context)` is better, update it, thanks!

Reply:
to be honest, I don't see any difference between current and original code?

in both cases get_context is used only when device is GPU